### PR TITLE
add approximate memtable stats

### PR DESCRIPTION
**Pull request by huachaohuang**

### CODE REVIEW

File: librocksdb_sys/src/lib.rs [hunk 1]
```diff
@@ -688,6 +688,21 @@ extern "C" {
                                          range_limit_key: *const *const u8,
                                          range_limit_key_len: *const size_t,
                                          sizes: *mut uint64_t);
+    pub fn crocksdb_approximate_memtable_stats(db: *mut DBInstance,
```
Comment (BusyJay) [inline librocksdb_sys/src/lib.rs]: Why mut?

File: librocksdb_sys/src/lib.rs [hunk 2]
```diff
@@ -688,6 +688,21 @@ extern "C" {
...
+                                               range_start_key: *const u8,
+                                               range_start_key_len: size_t,
+                                               range_limit_key: *const u8,
+                                               range_limit_key_len: size_t,
+                                               count: *mut uint64_t,
+                                               size: *mut uint64_t);
+    pub fn crocksdb_approximate_memtable_stats_cf(db: *mut DBInstance,
+                                                  cf: *mut DBCFHandle,
```
Comment (BusyJay) [inline librocksdb_sys/src/lib.rs]: Why mut?

File: src/rocksdb.rs [hunk 3]
```diff
@@ -1927,4 +1956,34 @@ mod test {
...
     }
+
+    #[test]
+    fn test_get_approximate_memtable_stats() {
+        let mut opts = DBOptions::new();
+        opts.create_if_missing(true);
+        let path = TempDir::new("_rust_rocksdb_get_approximate_memtable_stats").expect("");
+        let db = DB::open(opts, path.path().to_str().unwrap()).unwrap();
+
+        let samples = vec![(b"key1".to_vec(), b"value1".to_vec()),
```
Comment (BusyJay) [inline src/rocksdb.rs]: Why `to_vec`?

File: librocksdb_sys/crocksdb/crocksdb/c.h [hunk 4]
```diff
@@ -370,6 +370,18 @@ extern C_ROCKSDB_LIBRARY_API void crocksdb_approximate_sizes_cf(
...
     const size_t* range_limit_key_len, uint64_t* sizes);
 
+extern C_ROCKSDB_LIBRARY_API void crocksdb_approximate_memtable_stats(
+    crocksdb_t* db,
+    const char* range_start_key, size_t range_start_key_len,
+    const char* range_limit_key, size_t range_limit_key_len,
+    uint64_t* count, uint64_t* size);
+
+extern C_ROCKSDB_LIBRARY_API void crocksdb_approximate_memtable_stats_cf(
+    crocksdb_t* db, crocksdb_column_family_handle_t* column_family,
```
Comment (BusyJay) [inline librocksdb_sys/crocksdb/crocksdb/c.h]: Not consistent with the librocksdb_sys/src/lib.rs

File: src/rocksdb.rs [hunk 5]
```diff
@@ -820,6 +820,35 @@ impl DB {
         sizes
     }
 
+    pub fn get_approximate_memtable_stats(&self, range: &Range) -> (u64, u64) {
```
Comment (siddontang) [inline src/rocksdb.rs]: add an explanation for the return type, or define a typedef for the tuple.

**Pull request merged**